app/kubernetes/components/kube-services/kube-services-item/kube-services-item.controller.js

```diff
@@ -0,0 +1,45 @@
+import { KubernetesServicePort, KubernetesIngressServiceRoute } from 'Kubernetes/models/service/models';
+import { KubernetesFormValidationReferences } from 'Kubernetes/models/application/formValues';
+
+export default class KubeServicesItemViewController {
+  /* @ngInject */
+  constructor($async) {
+    this.$async = $async;
+  }
+
+  addPort() {
+    const p = new KubernetesServicePort();
+    p.nodePort = '';
+    p.port = '';
+    p.targetPort = '';
+    p.protocol = 'TCP';
+
+    if (this.ingressType) {
+      const r = new KubernetesIngressServiceRoute();
+      r.ServiceName = this.serviceName;
+      p.ingress = r;
+      p.Ingress = true;
+      this.servicePorts.push(p);
+    } else {
+      this.servicePorts.push(p);
```

Comment:
servicePorts.push lines are duplicated. Could be simplified.

Reply:
yes. fixed